src/main/java/com/sprint/mission/discodeit/storage/S3BinaryContentStorage.java

```diff
@@ -0,0 +1,149 @@
+package com.sprint.mission.discodeit.storage;
+
+import com.sprint.mission.discodeit.dto.binaryContent.BinaryContentDto;
+import com.sprint.mission.discodeit.exception.file.FileNotFoundCustomException;
+import com.sprint.mission.discodeit.exception.file.FileReadFailedException;
+import com.sprint.mission.discodeit.exception.file.FileSaveFailedException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.NoSuchKeyException;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.presigner.S3Presigner;
+import software.amazon.awssdk.services.s3.presigner.model.GetObjectPresignRequest;
+
+import java.io.InputStream;
+import java.time.Duration;
+import java.util.UUID;
+
+
+@Slf4j
+@Component
+@ConditionalOnProperty(name = "discodeit.storage.type", havingValue = "s3")
+public class S3BinaryContentStorage implements BinaryContentStorage {
+
+    private final String bucket;
+    private final S3Client s3Client;
+    private final S3Presigner presigner;
+
+    // 환경 변수에서 URL 만료 시간을 문자열로 받아서 Duration으로 변환
+    private final Duration expiration;
+
+    /**
+     * 생성자에서 S3Client, Presigner 초기화
+     */
+    public S3BinaryContentStorage(
+            @Value("${AWS_S3_ACCESS_KEY}") String accessKey,
+            @Value("${AWS_S3_SECRET_KEY}") String secretKey,
+            @Value("${AWS_S3_REGION}") String region,
+            @Value("${AWS_S3_BUCKET}") String bucket
```

Comment:
```suggestion
            @Value("${discodeit.storage.s3.access-key}") String accessKey,
            @Value("${discodeit.storage.s3.secret-key}") String secretKey,
            @Value("${discodeit.storage.s3.region}") String region,
            @Value("${discodeit.storage.s3.bucket}") String bucket
```

운영체제 환경변수가 아닌 spring properties로 하는게 더 적절해 보입니다!